packages/camera/camera_android_camerax/android/src/test/java/io/flutter/plugins/camerax/PreviewTest.java

```diff
@@ -0,0 +1,177 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camerax;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import android.graphics.SurfaceTexture;
+import android.util.Size;
+import android.view.Surface;
+import androidx.camera.core.Preview;
+import androidx.camera.core.SurfaceRequest;
+import androidx.core.util.Consumer;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.ResolutionInfo;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.SystemServicesFlutterApi.Reply;
+import io.flutter.view.TextureRegistry;
+import java.util.concurrent.Executor;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.robolectric.RobolectricTestRunner;
+
+@RunWith(RobolectricTestRunner.class)
+public class PreviewTest {
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Mock public Preview mockPreview;
+  @Mock public BinaryMessenger mockBinaryMessenger;
+  @Mock public TextureRegistry mockTextureRegistry;
+  @Mock public CameraXProxy mockCameraXProxy;
+
+  InstanceManager testInstanceManager;
+
+  @Before
+  public void setUp() {
+    testInstanceManager = spy(InstanceManager.open(identifier -> {}));
+  }
+
+  @After
+  public void tearDown() {
+    testInstanceManager.close();
+  }
+
+  @Test
+  public void createTest() {
+    final PreviewHostApiImpl previewHostApi =
+        new PreviewHostApiImpl(mockBinaryMessenger, testInstanceManager, mockTextureRegistry);
+    final Preview.Builder mockPreviewBuilder = mock(Preview.Builder.class);
+    final GeneratedCameraXLibrary.ResolutionInfo resolutionInfo =
+        new GeneratedCameraXLibrary.ResolutionInfo.Builder().setWidth(10L).setHeight(50L).build();
+
+    previewHostApi.cameraXProxy = mockCameraXProxy;
+    when(mockCameraXProxy.createPreviewBuilder()).thenReturn(mockPreviewBuilder);
+    when(mockPreviewBuilder.build()).thenReturn(mockPreview);
+
+    final ArgumentCaptor<Size> sizeCaptor = ArgumentCaptor.forClass(Size.class);
+
+    previewHostApi.create(3L, 90L, resolutionInfo);
+
+    verify(mockPreviewBuilder).setTargetRotation(90);
+    verify(mockPreviewBuilder).setTargetResolution(sizeCaptor.capture());
+    assertEquals(sizeCaptor.getValue().getWidth(), 10);
+    assertEquals(sizeCaptor.getValue().getHeight(), 50);
```

Comment:
Please put all constants that are expected to be the same into named variables. 
Ex. 
```
final long previewHeight = 50L; 
... .setHeight(50L).build()
assertEquals(sizeCaptor.getValue().getHeight(), previewHeight);
```

This makes the code much easier to read and requires the reader to not have to search back up the test to see where the values came from.

Reply:
For tests, I consider it better practice to use literals when possible. I think it makes tests a lot more readable. This is also the pattern that I have seen used most throughout the repo with every language.

Reply:
We discussed this in the plugins sync yesterday and came to the conclusion that when a function manipulates data and then you "know" the output using literals is a good practice BUT in the majority of the cases in this PR we are setting data and ensuring that the returned data is the same. In these situations locally defined constants are prefered. Please feel free to update this to use constants.